server/service/device_client.go

```diff
@@ -91,3 +80,41 @@ func (dc *DeviceClient) Ping() error {
 
 	return err
 }
+
+func (dc *DeviceClient) getListDevicePolicies(token string) ([]*fleet.HostPolicy, error) {
+	verb, path := "GET", "/api/latest/fleet/device/"+token+"/policies"
+	var responseBody listDevicePoliciesResponse
+	err := dc.request(verb, path, "", &responseBody)
+	return responseBody.Policies, err
+}
+
+func (dc *DeviceClient) getMinDesktopPayload(token string) (fleetDesktopResponse, error) {
+	verb, path := "GET", "/api/latest/fleet/device/"+token+"/desktop"
+	var r fleetDesktopResponse
+	err := dc.request(verb, path, "", &r)
+	return r, err
+}
+
+func (dc *DeviceClient) NumberOfFailingPolicies(token string) (uint, error) {
+	r, err := dc.getMinDesktopPayload(token)
+	if err == nil {
+		return uintValueOrZero(r.FailingPolicies), nil
+	}
+
+	if errors.Is(err, notFoundErr{}) {
```

Comment:
Were you able to test/QA this code branch with an old Fleet server (or one where you comment out the new endpoint)?

Reply:
Without lightweight endpoint:
![Screenshot from 2022-10-12 15-46-59](https://user-images.githubusercontent.com/8041318/195436190-e70c6f46-73d1-4e07-abe2-27a3fd16898c.png)

With lightweight endpoint:
![Screenshot from 2022-10-12 15-44-53](https://user-images.githubusercontent.com/8041318/195436294-cb21260f-6b24-4bb0-b49d-30fac62223d9.png)